File: WorkFlows/Chapter03/CIfElseSequentialExample/Workflow1.cs
using System;
using System.ComponentModel;
using System.ComponentModel.Design;
using System.Collections;
using System.Drawing;
using System.Workflow.ComponentModel.Compiler;
using System.Workflow.ComponentModel.Serialization;
using System.Workflow.ComponentModel;
using System.Workflow.ComponentModel.Design;
using System.Workflow.Runtime;
using System.Workflow.Activities;
using System.Workflow.Activities.Rules;

namespace CIfElseSequentialExample
{
	public sealed partial class Workflow1: SequentialWorkflowActivity
	{
        private int IntInputValue;

        public int InputValue
        {
            set
            {
               IntInputValue  = value;
            }
        }
 
		public Workflow1()
		{
			InitializeComponent();
		}

        private void Branch1Condition(object sender, ConditionalEventArgs e)
        {
            e.Result = IntInputValue > 50;
        }

        private void Branch2Condition(object sender, ConditionalEventArgs e)
        {
            e.Result = IntInputValue > 25;
        }

        private void Branch1Code_ExecuteCode(object sender, EventArgs e)
        {
            Console.WriteLine("Branch1");
        }

        private void Branch2Code_ExecuteCode(object sender, EventArgs e)
        {
            Console.WriteLine("Branch2");
        }
	}

}

File: WorkFlows/Chapter03/VBIfElseSequentialExample/Workflow1.vb
Imports System
Imports System.ComponentModel
Imports System.ComponentModel.Design
Imports System.Collections
Imports System.Drawing
Imports System.Workflow.ComponentModel.Compiler
Imports System.Workflow.ComponentModel.Serialization
Imports System.Workflow.ComponentModel
Imports System.Workflow.ComponentModel.Design
Imports System.Workflow.Runtime
Imports System.Workflow.Activities
Imports System.Workflow.Activities.Rules

'NOTE: When changing the namespace; please update XmlnsDefinitionAttribute in AssemblyInfo.vb
Partial Public class Workflow1
    Inherits SequentialWorkflowActivity
    Private IntInputValue As Integer
    Public WriteOnly Property InputValue() As Integer
        Set(ByVal value As Integer)
            IntInputValue = value
        End Set
    End Property
    Public Sub New()
        MyBase.New()
        InitializeComponent()

    End Sub
   
    Private Sub Branch1Code_ExecuteCode(ByVal sender As System.Object, ByVal e As System.EventArgs)
        MsgBox("Branch 1")
    End Sub

    Public Sub Branch1Condition(ByVal sender As Object, ByVal e As ConditionalEventArgs)
        e.Result = IntInputValue > 50
    End Sub
    Public Sub Branch2Condition(ByVal sender As Object, ByVal e As ConditionalEventArgs)
        e.Result = IntInputValue > 25
    End Sub

    Private Sub Branch2Code_ExecuteCode(ByVal sender As System.Object, ByVal e As System.EventArgs)
        MsgBox("Branch 2")
    End Sub
End Class
